languageserver/integration/commands.go

```diff
@@ -208,14 +210,70 @@ func (i *FlowIntegration) sendTransaction(conn protocol.Conn, args ...interface{
 		return nil, fmt.Errorf("failed to parse JSON arguments")
 	}
 
-	signer, err := i.state.EmulatorServiceAccount()
+	serviceAccount, err := i.state.EmulatorServiceAccount()
 	if err != nil {
 		return nil, fmt.Errorf("failed to get service account, err: %w", err)
 	}
-	signer.SetAddress(flow.HexToAddress(signers[0]))
 
-	const gasLimit uint64 = 1000
-	_, txResult, err := i.sharedServices.Transactions.Send(signer, code, "", gasLimit, txArgs, "")
+	payer := serviceAccount.Address()
+	keyIndex := serviceAccount.Key().Index()
+
+	// We need to check if service account among authorizers
+	hasServiceAccount := false
+
+	signerAccounts := make([]flowkit.Account, len(signers))
+	authorizers := make([]flow.Address, len(signers))
+	for i, v := range signers {
+		address := flow.HexToAddress(v)
```

Comment:
It would be better to change line 199 to convert to `flow.Address` instead of string so you don't have to then convert string to address in multiple places.